x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/pick_fields_step/components/advanced_detector_modal/advanced_detector_modal.tsx

```diff
@@ -0,0 +1,308 @@
+  const [overFieldOption, setOverFieldOption] = useState(createFieldOption(detector.overField));
+  const [partitionFieldOption, setPartitionFieldOption] = useState(
+    createFieldOption(detector.partitionField)
+  );
+  const [excludeFrequentOption, setExcludeFrequentOption] = useState(
+    createExcludeFrequentOption(detector.excludeFrequent)
+  );
+  const [fieldsEnabled, setFieldsEnabled] = useState(true);
+  const [fieldOptionEnabled, setFieldOptionEnabled] = useState(true);
+
+  const aggOptions: EuiComboBoxOptionProps[] = aggs.map(createAggOption);
+  // .sort((a, b) => a.label.localeCompare(b.label));
+  const fieldOptions: EuiComboBoxOptionProps[] = fields
+    .filter(f => f.id !== EVENT_RATE_FIELD_ID)
+    .map(createFieldOption);
+  const splitFieldOptions = [...fieldOptions, ...createMlcategoryField(jobCreator)];
+
+  const eventRateField = fields.find(f => f.id === EVENT_RATE_FIELD_ID);
+
+  const onOptionChange = (func: (p: EuiComboBoxOptionProps) => any) => (
+    selectedOptions: EuiComboBoxOptionProps[]
+  ) => {
+    func(selectedOptions[0] || emptyOption);
+  };
+
+  function getAgg(title: string) {
+    return aggs.find(a => a.id === title) || null;
+  }
+  function getField(title: string) {
+    if (title === mlCategory.id) {
+      return mlCategory;
+    }
+    return fields.find(a => a.id === title) || null;
+  }
+
+  useEffect(() => {
+    const agg = getAgg(aggOption.label);
+    let field = getField(fieldOption.label);
+
+    if (agg !== null) {
+      setFieldsEnabled(true);
+      if (isCountAgg(agg) && eventRateField !== undefined) {
+        setFieldOption(emptyOption);
+        setFieldOptionEnabled(false);
+        field = eventRateField;
+      } else {
+        setFieldOptionEnabled(true);
+      }
+    } else {
+      setFieldsEnabled(false);
+    }
+
+    const dtr: RichDetector = {
+      agg,
+      field,
+      byField: getField(byFieldOption.label),
+      overField: getField(overFieldOption.label),
+      partitionField: getField(partitionFieldOption.label),
+      excludeFrequent: excludeFrequentOption.label !== '' ? excludeFrequentOption.label : null,
+    };
+    setDetector(dtr);
+  }, [
+    aggOption,
+    fieldOption,
+    byFieldOption,
+    overFieldOption,
+    partitionFieldOption,
+    excludeFrequentOption,
+  ]);
+
+  useEffect(() => {
+    const agg = getAgg(aggOption.label);
+    setFieldsEnabled(aggOption.label !== '');
+    if (agg !== null) {
+      setFieldOptionEnabled(isCountAgg(agg) === false);
+    }
+  }, []);
+
+  function onCreateClick() {
+    detectorChangeHandler(detector, payload.index);
+  }
+
+  function saveEnabled() {
+    return (
+      fieldsEnabled &&
+      (fieldOptionEnabled === false || (fieldOptionEnabled === true && fieldOption.label !== ''))
+    );
+  }
+
+  return (
+    <ModalWrapper onCreateClick={onCreateClick} closeModal={closeModal} saveEnabled={saveEnabled()}>
```

Comment:
I can't see any way to edit the description of the detector? Is it hidden away somewhere? It used to be at the top of the Add detector modal.

Reply:
Will add in a follow up PR.
https://github.com/elastic/kibana/projects/30#card-27607228